presto-main/src/main/java/com/facebook/presto/operator/aggregation/histogram/HashUtil.java

```diff
@@ -0,0 +1,73 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.operator.aggregation.histogram;
+
+import static com.google.common.base.Preconditions.checkArgument;
+import static it.unimi.dsi.fastutil.HashCommon.arraySize;
+
+public class HashUtil
+{
+    private HashUtil() { }
+
+    public static int nextProbeLinear(int probeCount)
+    {
+        return probeCount;
+    }
+
+    public static int nextSumOfCount(int probeCount)
+    {
+        return (probeCount * (probeCount + 1)) / 2;
+    }
+
+    public static int nextSumOfSquares(int probeCount)
+    {
+        return (probeCount * (probeCount * probeCount + 1)) / 2;
```

Comment:
This is doing cubic-like probe, right? :)

Reply:
yup--special case of cubic w/o the constants (or their 0)

JMH showed up to a 10% win sometimes, other times a wash over linear. we have methods for all 3 (linear, quadratic, cubic) and can just make a code change

the JMH results seemed convincing enough to go with this one

Reply:
https://en.wikipedia.org/wiki/Quadratic_probing

i played with the variations, but found doing the sum of probe_count and sum of probe_count^2 to perform best.

given (yet more) time I'd dig into it, but this was mostly just an incidental gain